### PR TITLE
Organize makefile

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
- Organize Makefile by section to be easier to find where it any kind of target and know ts purpose. 

Screenshot 2019-11-17 at 10 16 30

NOTE: Not merge this one before https://github.com/operator-framework/operator-sdk/pull/2205

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,6 +68,12 @@ tidy: ## Update dependencies
 clean: ## Clean up the build artifacts
 	$(Q)rm -rf build
 
+##############################
+# Generate Artefacts         #
```
Comment (jmrodri) [inline Makefile]: Typo: Artefacts should be Artifacts

Comment (camilamacedo86) [inline Makefile]:
```suggestion
# Generate Artifacts         #
```

File: Makefile [hunk 2]
```diff
@@ -63,19 +83,15 @@ gen-test-framework: install ## Run generate commands to update test/test-framewo
...
 .PHONY: generate gen-cli-doc gen-test-framework
 
+##############################
+# Release                    #
+##############################
+
+##@ Release
+
 # Build/install/release the SDK.
 .PHONY: install release_builds release
```
Comment (estroz) [inline Makefile]:
```suggestion
.PHONY: release_builds release
```

File: Makefile [hunk 3]
```diff
@@ -35,13 +35,27 @@ help: ## Show this help screen
...
-	@echo ''
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+##############################
+# Development                #
+##############################
+
+##@ Development
 
 .PHONY: all
```
Comment (estroz) [inline Makefile]:
```suggestion
.PHONY: all install

```

**Pull request merged**